File: frontend/app/src/app/model/Member.ts
export type MemberType = 'GLOBAL' | 'SITE' | 'FREE';

export interface Member {
  id: string;
  matricule: string;
  name: string;
  type: 'GLOBAL' | 'SITE' | 'FREE';
  siteId?: string | null;
  unpaidBalance: number;
  blockedUntil?: string | null;
  adminRole?: 'NONE' | 'GLOBAL_ADMIN' | 'SITE_ADMIN';
}

File: backend/src/main/java/be/ephec/pdw/backend/member/Member.java
package be.ephec.pdw.backend.member;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Member {

    @Id
    private UUID id;

    @Column(nullable = false, unique = true)
    private String matricule;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MemberType type;

    private String siteId;

    @Column(nullable = false)
    private BigDecimal unpaidBalance;

    private LocalDate blockedUntil;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private AdminRole adminRole = AdminRole.NONE;

    private String adminPassword;
}